library/std/src/sys/unix/thread_local_dtor.rs

```diff
@@ -16,23 +16,36 @@ pub unsafe fn register_dtor(t: *mut u8, dtor: unsafe extern "C" fn(*mut u8)) {
     use crate::mem;
     use crate::sys_common::thread_local_dtor::register_dtor_fallback;
 
+    /// This is necessary because the __cxa_thread_atexit_impl implementation
+    /// std links to by default may be a C or C++ implementation that was not
+    /// compiled using the Clang integer normalization option.
+    #[cfg(not(sanitizer_cfi_normalize_integers))]
+    #[cfi_encoding = "i"]
+    #[repr(transparent)]
+    pub struct c_int(pub libc::c_int);
+
     extern "C" {
         #[linkage = "extern_weak"]
         static __dso_handle: *mut u8;
         #[linkage = "extern_weak"]
-        static __cxa_thread_atexit_impl: *const libc::c_void;
+        fn __cxa_thread_atexit_impl(
+            dtor: unsafe extern "C" fn(*mut libc::c_void),
+            arg: *mut libc::c_void,
+            dso_handle: *mut libc::c_void,
+        ) -> c_int;
```

Comment:
I believe what you want is
```suggestion
        static __cxa_thread_atexit_impl: Option<unsafe extern "C" fn(
            dtor: unsafe extern "C" fn(*mut libc::c_void),
            arg: *mut libc::c_void,
            dso_handle: *mut libc::c_void,
        ) -> c_int>;
```